Management/Clients/AirQualityDataClient.cs

```diff
@@ -45,7 +54,101 @@ public async Task<AirQualityCityResponse> GetAirQualityAsync(string city, string
             throw new Exception("Retrieve Data Failed");
         }
 
-        public int CalculateScoreAsync()
-            => throw new NotImplementedException();
+        // Use Air Quality Index
+        // https://www.airnow.gov/aqi/aqi-basics/
+        public async Task<Dictionary<City, int>> CalculateScoresAsync(
+            IEnumerable<City> cities,
+            State state,
+            CountryCode countryCode,
+            CancellationToken cancellationToken)
+        {
+            var bag = new ConcurrentBag<(City city, int score)>();
```

Comment:
ConcurrentBag just like List 
But this is for Parllel execution, it will prevent the race conditon. You cannot assume the result is in order as well.